docs/dependency-updates.md

```diff
@@ -0,0 +1,164 @@
+## npm
+
+We have two npm projects in this repository: the website frontend (identified by
+`/frontend/package.json`), and the root project from which our end-to-end tests
+are run (identified by `/package.json`). The former is defined as a
+[workspace](https://docs.npmjs.com/cli/v8/using-npm/workspaces) in the latter,
+and both have their dependencies checked by Dependabot.
+
+What follows is a list of dependencies and how to check for potential breakage when they release new versions:
+
+### `*lint*`, `@types/*`, `typescript`, `*jest*`, `@testing-library/*`, `react-test-renderer`, `fast-check`, `prettier`
+
+For linters, type definitions, and non-Playwright testing-related packages, a
+successful CI run generally provides enough confidence that the upgrade is fine.
+That said, it can't harm to occassionally scan the results of the
+`build_frontend` or `test_frontend` CI runs to see if any new warnings were
+added. If `jest-junit` is updated, you'll want to check that CircleCI can still
+see code coverage for the frontend.
+
+### `@playwright/test` and `dotenv`
+
+Our Playwright tests are currently only triggered in CI once a day from the
+`main` branch. Thus, there's unfortunately no easy way to verify that it still
+runs successfully until after merging. To do so, after merging the PR, open the
+"Actions" tab in GitHub, then find the "[Relay e2e
+Tests](https://github.com/mozilla/fx-private-relay/actions/workflows/playwright.yml)"
+workflow on the left-hand side, and then use the "Run workflow" trigger button
+in the top row of the table.
+
+`dotenv` is used to load environment variables for Playwright from `.env` files,
+so can be verified in the same way.
+
+### `react-aria` and `react-stately`
+
+These dependencies handle most of the dynamic UI elements in our frontend, such
+as the user menu and the app menu in the header, the modals when claiming a
+subdomain or creating a new custom mask, or the toggle that allows you to set a
+mask to block nothing, promotional emails, or everything. To verfy these
+updates, open the preview deployments and check that you can still interact with
+those using both a keyboard and a mouse.
+
+### `@fluent/*`
+
+[Fluent](https://projectfluent.org/) handles localisation, so verifying these
+involves setting your browser language to something other than English, and
+verifing that the website content indeed shows up in your chosen different
+language.
+
+### `next`
+
+[Next.js](https://nextjs.org/) handles most of our frontend build pipeline and
+bundling the website. To spot-check this update, open the Preview deployment to
+verify that the website was put together successfully, and that you can navigate
+between different pages (e.g. from the home to the FAQ) without issues.
+
+### `@next/bundle-analyzer`
+
+We use this package to investigate if any of our dependencies take up an
+inordinate amount of bytes in our payload. To verify that this still runs
+successfully, run
+
+    ANALYZE=true npm run build
```

Comment:
The thing is that there's not really a pass or fail result; it rather gives you a report of which modules take up which percentage of the final bundle. So it's a tool to help investigate in case issues with our bundle size occur. (I think there's also something to be said for removing this dependency, since I don't think we've used it more than once or twice - so we can just temporarily install it if we need it in the future.)

Reply:
I see. Looks like we could run the underlying `webpack-bundle-analyzer` tool with `-m json` and then make sure the total size is under a certain value. But again - outside the scope of this PR.